src/invoice/interfaces.ts

```diff
@@ -0,0 +1,98 @@
+export interface CardDetails {
+    Brand: string,
+    LastDigits: string,
+    ExpirationMonth: number,
+    ExpirationYear: number,
+    CardId: string // Square/Stripe Card ID
+}
+
+export interface InvoicePayment {
+    Date: number, //Unix Timestamp
+    Amount: number,
+    Currency: string,
+    Method: 'Stripe' | 'CreditCard' | 'Cash' | 'Check' | 'DebitCard' | 'Square' | 'CreditBalance' | 'BankTransfer' | 'Insurance' | 'GiftCard' | 'Other'
+    ProcessedBy: string,
+    ProcessedByType: 'Client' | 'Provider' | 'Automation',
+    TipAmount: number,
+    RefundedAmount: number,
+    AdditionalInfo: string,
+    CardDetails: CardDetails
+}
+
+export interface InvoiceSubItem {
+    ServiceCode: number,
+    Price: number,
+    Units: number,
+    Description: string,
+    Date: number,
+    Modifiers: any[] // They don't actually ever mention what this is...
+}
+
+export interface InvoiceItem {
+    AppointmentId: string,
+    ProductId: string,
+    Description: string,
+    TotalTaxAmount: number,
+    Units: number,
+    Price: number,
+    Date: number, // Unix Timestamp
+    Taxes: { Name: string, Percentage: number }[],
+    TaxesIncludedInPrice: boolean,
+    IsCopay: boolean,
+    TotalAmount: number,
+    ServiceCode: number,
+    SubItems: InvoiceSubItem[]
+}
+
+export interface Invoice {
+    // These two fields seem to be the same, just one is a number and one is a string for some reason...
```

Comment:
Probably same as the Client Api having `ClientId` and `ClientNumber`

Reply:
Oh good.